File: python_benchmark/Kruskal_.py
import numpy as np
import pandas as pd
from scipy.stats import kruskal

def Kruskal(random_arr):
    arr_num = random_arr.shape[0]
    arr_size = random_arr.shape[1]
    kruskal_res = np.zeros(int(arr_num/2))

    print("Start Kruskal test...")
    start = pd.Timestamp.now()
    for i in range (0,arr_num-2,2):
        # 将两个样本的数据合并成一个数组，并添加一个指示变量
        data = np.concatenate([random_arr[i], random_arr[i+1]])
        groups = np.concatenate([np.zeros(arr_size), np.ones(arr_size)])
        # 进行Kruskal-Wallis检验
        kruskal_res[int(i/2)] = kruskal(*[data[groups == k] for k in np.unique(groups)])[0]
    print("Kruskal shape:", kruskal_res.shape)
    end = pd.Timestamp.now()
    print('Kruskal time cost: ', end - start)

File: matlab benchmark/seq/Kruskal.m
function p_values = Kruskal(random_arr)
    % 计算Kruskal两变量检验
    arr_num = size(random_arr, 1);
    fprintf("size of random_arr: %d\n", arr_num);
    % p_values = zeros(arr_num, arr_num);
    p_values = zeros(arr_num/2);

    disp('start kruskalwallis test');
    
    tic
    for i = 1:2:arr_num-1
            [p, ~, ~] = kruskalwallis([random_arr(i,:); random_arr(i+1,:)], [], 'off');
            p_values(ceil(i/2)) = p;
    end
    toc;

    fprintf("size of p_values: %d\n", size(p_values, 1));
    fprintf('end kruskalwallis test\n\n');

    end